lib/git-fastclone.rb

```diff
@@ -338,6 +357,31 @@ def store_updated_repo(url, mirror, repo_name, fail_hard)
       raise e if fail_hard
     end
 
+    def retriable_error?(error)
+      error_strings = [
+        'fatal: missing blob object',
+        'fatal: remote did not send all necessary objects',
+        /fatal: packed object [a-z0-9]+ \(stored in .*?\) is corrupt/,
+        /fatal: pack has \d+ unresolved delta/,
+        'error: unable to read sha1 file of ',
+        'fatal: did not receive expected object',
+        /^fatal: unable to read tree [a-z0-9]+\n^warning: Clone succeeded, but checkout failed/
+      ]
+      error.to_s =~ /^STDERR:\n.*^#{Regexp.union(error_strings)}/m
```

Comment:
One minor logic change here when extracting this out to a method. I changed the `.+` to a `.*` here. It doesn't seem like there is any good reason to require something between the `STDERR:` and the patterns we are expecting to match. While I was cleaning up the specs, they were failing on some of the patterns above if they directly followed the `STDERR:` line.

Reply:
Makes sense to me, looking at the history of this file I think it was previously matching patterns after `fatal:` only and not `STDERR` with anything after it (e.g. `error:`). So probably a change that could've been made when the non `fatal:` patterns were introduced.

Reply:
I think it actually broke when I refactored the tests to test pattern matching in isolation, and I dropped an extra newline between the STDERR: and the actual error message. This new pattern is more lenient, so seems reasonable, but I could also alter the format_error method to match the old behavior.